sdk/src/connections/target/adsd3500_sensor.cpp

```diff
@@ -297,12 +292,40 @@ aditof::Status Adsd3500Sensor::open() {
             return Status::GENERIC_ERROR;
         }
 
-        dev->sfd = ::open(subDevName, O_RDWR | O_NONBLOCK, 0);
+        dev->sfd = ::open(subDevName, O_RDWR | O_NONBLOCK);
         if (dev->sfd == -1) {
             LOG(WARNING) << "Cannot open " << subDevName << " errno: " << errno
                          << " error: " << strerror(errno);
             return Status::GENERIC_ERROR;
         }
+
+        //Check chip status and reset if there are any errors.
+        if (m_firstRun) {
+            aditof::Status chipStatus = aditof::Status::GENERIC_ERROR;
+            uint16_t chipID;
+            uint8_t dealiasCheck[32] = {0};
+            dealiasCheck[0] = 1;
+
+            while (chipStatus != aditof::Status::OK) {
+                chipStatus = adsd3500_read_cmd(0x0112, &chipID);
+                if (chipStatus != Status::OK) {
+                    LOG(ERROR)
+                        << "Failed to read adsd3500 chip id! Reseting chip...";
+                    adsd3500_reset();
+                    continue;
+                }
+
+                chipStatus = adsd3500_read_payload_cmd(0x02, dealiasCheck, 32);
+                if (chipStatus != Status::OK) {
+                    LOG(ERROR) << "Failed to read dealias parameters for "
+                                  "adsd3500. Reseting chip...";
+                    adsd3500_reset();
+                }
+            }
+            m_firstRun = false;
+
+            LOG(INFO) << "Chip programmed successfully!";
```

Comment:
I would rephrase this since it sounds like a programming just happened. Maybe say something like:
"ADSD3500 is ready to communicate with"